### DISCUSSION TITLE
Resource Block allocation in UE

### DISCUSSION THREAD

**Discussion by warlock1302** [srsran/srsRAN_Project#689] [General Help] [answered]
Hello,
As part of our research, we need the information on the Resource Blocks(RB) assigned to a UE. This can be either in the form of a bitmap, or pairs of (Start_RB, Num_RBs) depending on the resource allocation type.
This is typically carried by the PDCCH. Is this feasible to collect in srsUE?
Our Goal:
To learn the resource allocation patterns for Machine learning and devise a targeted jamming plan (e.g., jam RBs that are allocated to different UEs and maximise the disruption, rather than jamming randomly)
Please let us know if this information can be collected.

Thanks,
Abhishek VJ
Reasearch Assistant
Techniche Universitat Chemnitz

**Comment by herlesupreeth** [comment]
> As part of our research, we need the information on the Resource Blocks(RB) assigned to a UE. This can be either in the form of a bitmap, or pairs of (Start_RB, Num_RBs) depending on the resource allocation type.
> This is typically carried by the PDCCH. Is this feasible to collect in srsUE?

You can get these in gNB logs as well if set MAC log level to info/debug. Currently, we support only resource allocation type 1

**Reply by warlock1302** [reply]
@herlesupreeth thanks for your quick reply. i have collected the gNB logs with debug enabled but i could not locate the resource block information or i guess i do not know how to read them. could you please help point out the exact parameter i have to check.
i have attached the gnb log for your reference. Thanks
[gnb_1.log](https://github.com/user-attachments/files/15973589/gnb_1.log)

**Reply by herlesupreeth** [reply]
In case of PDSCH its the value in `rb=[...)`, e.g. rb=[0..40) is interpreted as VRBs between 0 and 40 and inclusive of only VRB 0

`- UE PDSCH: ue=0 c-rnti=0x4601 h_id=0 rb=[0..40) symb=[2..14) tbs=325 mcs=4 rv=0 nrtx=0 k1=4 olla=0.00 grants: lcid=62: size=6, lcid=0: size=315`

similarly for PUSCH you will find such logs below also having `rb=[...)` e.g. rb=[3..39) is interpreted as VRBs between 3 and 39 and inclusive of only VRB 3

`- UE PUSCH: ue=0 c-rnti=0x4601 h_id=0 rb=[3..39) symb=[0..14) tbs=512 rv=0 nrtx=0 olla=0.00 k2=4`

**Comment by warlock1302** [comment]
@herlesupreeth thanks a lot. it solved my problem.

i have a small query about the debug logs:

DL-PHY2 - Indicates that the log entry is from the Downlink Physical Layer.
But, as I go through the different log entries I can see different numbers, i.e., [DL-PHY2 ],[DL-PHY3 ], [DL-PHY4 ], etc.
Could you let me know what these numbers mean?

Also, the slot number is in a decimal format, e.g., 35.4. Can you please mention what does this means? I can see for example 125.5, 125.6, etc (and I once noticed a duplicate entry, I saw slot 126.7 twice. I cannot see why)

**Comment by herlesupreeth** [comment, marked as answer]
> But, as I go through the different log entries I can see different numbers, i.e., [DL-PHY2 ],[DL-PHY3 ], [DL-PHY4 ], etc.
> Could you let me know what these numbers mean?

Multiple PHY threads. 

> Also, the slot number is in a decimal format, e.g., 35.4. Can you please mention what does this means?

35.4 represents a SFN number of 35 and slot number 4

> I can see for example 125.5, 125.6, etc (and I once noticed a duplicate entry, I saw slot 126.7 twice. I cannot see why)

Not sure what you are referring here. If you could paste the logs I can understand better
